core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -293,6 +293,16 @@ public int getValue() {
    */
   UfsFileStatus getFileStatus(String path) throws IOException;
 
+  /**
+   * Computes and returns a fingerprint for the path. The fingerprint is used to determine if two
+   * UFS files are identical. The fingerprint must be deterministic, and must not change if a
+   * file is only renamed (identical content and permissions).
```

Comment:
When is this method expected to return an invalid fingerprint, and is that case recoverable?

Reply:
I wanted any error to return the invalid fingerprint, instead of exception, since an invalid fingerprint will only produce false negatives when comparing fingerprints. That will not affect correctness, but potentially performance. What do you think?

Reply:
In that case, should the signature not include `throws IOException` to make this explicit?

Reply:
Yes, I updated it. Thanks!